packages/api/cms-api/src/user-permissions/entities/user-permission.entity.ts

```diff
@@ -0,0 +1,49 @@
+import { BaseEntity, Entity, PrimaryKey, Property } from "@mikro-orm/core";
+import { Field, ID, ObjectType, registerEnumType } from "@nestjs/graphql";
+import { v4 } from "uuid";
+
+export enum UserPermissionSource {
+    MANUAL = "MANUAL",
+    BY_RULE = "BY_RULE",
+}
+registerEnumType(UserPermissionSource, {
+    name: "UserPermissionSource",
+});
+
+@ObjectType()
+@Entity({ tableName: "CometUserPermission" })
+export class UserPermission extends BaseEntity<UserPermission, "id"> {
+    @Field(() => ID)
+    @PrimaryKey({ type: "uuid" })
+    id: string = v4();
+
+    @Property()
+    userId: string;
+
+    @Field(() => UserPermissionSource, { nullable: true })
+    source: UserPermissionSource;
```

Comment:
```suggestion
    source?: UserPermissionSource;
```

This is marked as nullable in the GraphQL schema

Reply:
74f7ec193f666a4c0ada35adfd1e67bbf540d462

Reply:
Johnny "Eagly Eye" Omair

Reply:
Sorry 🙈